### ISSUE TITLE
errors opening encrypted file: "spawnSync sops ENOENT"

### ISSUE THREAD

**Issue by Morriz** [signageos/vscode-sops#4]
I guess your extension is making a lot of assumptions which are not documented.

We have the extension `.enc` on encrypted files, and your extension detects it, and then throws this error:

```
Could not decrypt SOPS file ...: spawnSync sops ENOENT
```

**Comment by misak113** [comment]
Hi, the .enc extension of files should not be the problem.
The detection of a file type is based on internal VSCode API which returns the language of the file when opened. The only YAML and JSON languages are observed by extension.
If the file with the custom extension (.enc) is not detected as YAML file, even the syntax highlighting wouldn't work for you.

The problem could be related to some missing global requirements. Currently, there has to be installed Node.js and SOPS binaries and exported in PATH environment variable. See https://github.com/signageos/vscode-sops#requirements 
Could you print following:
```
env
whereis sops
whereis node
# or at least
echo $PATH
```

I consider a later version to be able to work without node.js. However, this first version is just PoC.

I welcome pull requests.

**Comment by Morriz** [comment]
I don't have whereis but I have both node and sops and use them already a lot. I am using GCP KMS though, and it needs GOOGLE_APPLICATION_CREDENTIALS to point to my account json file. It is probably the only variable that should be able to be set in your extension.

**Comment by Morriz** [comment]
FYI:
```
which node
which sops
```
give me
```
/Users/Morriz/.nvm/versions/node/v11.15.0/bin/node
/usr/local/bin/sops
```

**Comment by Morriz** [comment]
nevermind, I killed vscode, set it in the terminal and started vscode from there...env var is now set

**Comment by Morriz** [comment]
it's working now, albeit quite slow...but thanks a lot!

**Comment by dzivkovic6** [comment]
I have this issue too, I did try setting the keys and everything but it still fails. Could someone write something more on how to fix this?

**Comment by misak113** [comment]
The problem is the missing sops or node binaries in the PATH environment variable. Just check if it's there based on your system. Like Linux Ubuntu can have the sops inside `~/.local/bin/sops` so PATH should contain `export PATH=$PATH:~/.local/bin`. But it can differ for every system, user, or whatever. It applies to the node binary as well...
VScode has limited options for getting environment variables, so maybe `~/.bash_profile` or `~/.bashrc` are just not enough. You may have to add it to `/etc/environment` and reboot the system.

**Comment by misak113** [comment]
Where exactly is located your sops? (The absolute path). Where is extended the PATH env var with the dirname of sops location? ~/.bashrc, ~/.zshrc, ~/.bash_profile or /etc/environment or other?
You can optionally specify the sops binary location in VSCode settings if nothing else works for you.
<img width="1186" alt="image" src="https://user-images.githubusercontent.com/1695995/179685730-605ace9c-80e7-4faa-b489-65ed2563f6a3.png">

**Comment by hammady** [comment]
The problem appears to be related to the node binary. In my setup, I have node installed via nvm where nvm itself is set up through ~/.bashrc. As VS code does not load that on boot, it can't find the node binary.
I believe a temporary fix to this issue is to allow the extension to look up the node binary path from a user setting, same like the sops binary path.

**Comment by misak113** [comment]
The latest version os vscode-sops extension shoul not be dependent on node.js already. The underlying scripts were changed to shell native instead of node.js.
Can you try it now and confirm that? In case there is problem let's create another issue and we will figure it out.

**Comment by benwalio** [comment]
adding in my own 2c, when I populated the Bin Path with the output of `whereis sops` everything started working correctly

**Comment by LucianU** [comment]
I just encountered this issue as well.

First, I did what `benwalio` suggested and I added the path to the `sops` binary to Extension Settings.
This path is in `$PATH`, but for some reason, the extension doesn't see it.

After I did this, the error changed. It said it failed to find the data key. But it used the default macOS
path (`/Users/lucian/Library/Application | Support/sops/age/keys.txt`), instead of using the `SOPS_AGE_KEY_FILE` variable, which I have set.

To test whether `PATH` and `SOPS_AGE_KEY_FILE` are available to vscode, I ran in vscode's Terminal:

```bash
echo $PATH
echo $SOPS_AGE_KEY_FILE
```
